executor/src/witgen/jit/processor.rs

```diff
@@ -140,39 +141,62 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Processor<'a, T, FixedEv
 
         // TODO Tuning: If this fails (or also if it does not generate progress right away),
         // we could also choose a different variable to branch on.
-        let left_branch_code =
-            self.generate_code_for_branch(can_process.clone(), first_branch, branch_depth + 1)?;
-        let right_branch_code =
-            self.generate_code_for_branch(can_process, second_branch, branch_depth + 1)?;
-        let code = if left_branch_code == right_branch_code {
-            common_code.into_iter().chain(left_branch_code).collect()
-        } else {
-            common_code
+
+        let first_branch_result =
+            self.generate_code_for_branch(can_process.clone(), first_branch, branch_depth + 1);
+        let second_branch_result =
+            self.generate_code_for_branch(can_process, second_branch, branch_depth + 1);
+        // There is a recoverable error: It might be that a branch has conflicting
+        // constraints, which means we can just take the other branch without actually
+        // branching.
+        match (first_branch_result, second_branch_result) {
+            (Err(el), Err(er))
+                if el.reason == ErrorReason::ConflictingConstraints
```

Comment:
I'm not a big fan of ifs in match statements, because they make it very hard to see which cases are covered IMO. I think a nested match on `(el.reason, er.reason)` would be better and fix the comment below.